### ISSUE TITLE
[DataGrid] Filter column for null / not null cell values

### ISSUE THREAD

**Issue by OscarDL** [mui-org/material-ui-x#1957]
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

A filter option to filter columns for cells having any value at all would be great, and I'm pretty surprised it doesn't seem to exist - as far as I know. 

## Examples 🌈

Using inspect element:
![example](https://user-images.githubusercontent.com/27915933/123237063-9eae0e00-d4dd-11eb-8e83-9227284147af.png)

Use cases for this feature request: filter either the first 2 (or last 3) rows from this short example.
![usecase](https://user-images.githubusercontent.com/27915933/123239136-5db6f900-d4df-11eb-8d06-60d8c2339912.png)

## Motivation 🔦

Filter out values where rows from a particular column has a value at all, or doesn't have a value at all.

**Comment by m4theushw** [comment]
Thanks for this suggestion. We'll try to accommodate it in our next sprints. 

For now, you can [extend](https://material-ui.com/components/data-grid/filtering/#extend-filter-operators) the default filter operators adding a new one. Here's an example:

```jsx
const emailColumnType = {
  extendType: 'string',
  filterOperators: [
    ...getGridStringOperators(),
    {
      value: 'isNullOrNot',
      getApplyFilterFn: (filterItem) => {
        if (!filterItem.columnField || !filterItem.value || !filterItem.operatorValue) {
          return null;
        }
        return ({ value }) => {
          return filterItem.value === 'null' ? value == null : value != null;
        };
      },
      InputComponent: (props) => (
        <GridFilterInputValue
          select
          SelectProps={{
            native: true,
          }}
          {...props}
        >
          <option value="null">null</option>
          <option value="notNull">not null</option>
        </GridFilterInputValue>
      ),
    },
  ],
};

export default function Demo() {
  return (
    <div style={{ height: 400, width: '100%' }}>
      <DataGrid
        rows={rows}
        localeText={{
          filterOperatorIsNullOrNot: 'is null or not',
        }}
        columns={[
          { field: 'name', width: 150 },
          { field: 'email', type: 'email', width: 150 },
        ]}
        columnTypes={{ email: emailColumnType }}
      />
    </div>
  );
}
```

**Comment by oliviertassinari** [comment]
Almost a duplicate of #1165, but a bit differnet.

**Comment by OscarDL** [comment]
Hey there, thanks for the quick reply!
This workaround works great for my needs, so thanks a lot for that.

I still feel like a "native" / pre-made implementation would be great, perhaps to more people too - so I'm not going to close this feature request (unless you'd like me to).

PS: small nitpick, is there a way to make the filter effective right when selecting the operator? As seen on [this GIF](https://i.imgur.com/nmReiyj.gif), the null filter doesn't work unless I specifically apply not null, then null again. Thanks in advance!

**Comment by oliviertassinari** [comment]
@OscarDL It could be a bug, Do you have a codesandbox?

As a side note, I love how the composition API of the toolbar is paying out:

<img width="1285" alt="Capture d’écran 2021-06-24 à 20 36 19" src="https://user-images.githubusercontent.com/3165635/123315462-0c812680-d52c-11eb-823d-b99c995afaa6.png">

**Comment by OscarDL** [comment]
> @OscarDL It could be a bug, Do you have a codesandbox? Please open a new issue if you do.

Seems like I was able to reproduce it in [CodeSandbox](https://codesandbox.io/s/blissful-volhard-uyc9i) indeed, I'll open a new issue then. Thanks again!

**Comment by flaviendelangle** [comment]
https://github.com/mui-org/material-ui-x/pull/1997

@m4theushw I think your PR fixed this issue

**Comment by m4theushw** [comment]
Fixed by #1997
